### DISCUSSION TITLE
Can we get more complete documentation for nx-enforce-module-boundaries?

### DISCUSSION THREAD

**Discussion by Splaktar** [nrwl/nx#5039] [Feature Requests]
The only documentation that I was able to find was a short mention in https://nx.dev/latest/angular/structure/dependency-graph#circular-dependencies.

I found the source code [here](https://github.com/nrwl/nx/blob/master/packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts). But it would be nice to have some documentation on the site for how to configure it.

It seems like a lot of people are referring to the blog post ["How to programmatically enforce boundaries between applications and libraries in an Nx monorepo"](https://medium.com/showpad-engineering/how-to-programmatically-enforce-boundaries-between-applications-and-libraries-in-an-nx-monorepo-39bf8fbec6ba), but it would be nice to have a documentation page with some details on how to configure it and some examples.

**Comment by epikhighs** [comment]
I asked a similar question in this [stack overflow post](https://stackoverflow.com/questions/67067004/in-nx-libs-why-are-module-boundaries-defined-through-public-apis-only).  It seems to me the concept of module boundaries needs a better explanation.  In particular, when it's useful and when it might not be strictly necessary.  Basically, I'd like to understand the tradeoffs better and figure out a customized solution.
